xcarchive/xcarchive.go

```diff
@@ -39,3 +42,38 @@ func IsMacOS(archPath string) (bool, error) {
 
 	return exist, nil
 }
+
+func unwrapFileEmbeddedInXcarchiveDir(xcarchivePth, fileName string) (string, error) {
```

Comment:
this method gets file paths that are unzipped, this is not a logic strictly for xcarchives, this matching for any zip related logic, please remove xcarchive labels from everywhere in this case

Reply:
Similarly to unwrapFileEmbeddedInPayloadAppDir in ipa.go, the purpose of this method is not to grant a universal solution for unwrapping, but to have a solution for xcarhives/ipas, where the needed files are in given folders. For reducing code duplication, there should be a general method, and these methods should use that. What do you think?